### PR TITLE
Hide the configured header image in iframe embed mode

### LINKED ISSUE #2186: Hide the configured header image in iframe embed mode
## Problem

The shared layout renders the configured Header Image before all page content. Requests with `?iframe=true` still render and load that image.

Embed mode already removes the public menu and listing header. The remaining image duplicates host-site branding and uses iframe space.

The paid popup also sends the parent iframe to `/ticket/reserved` without `iframe=true`. The normal header can therefore return after payment.

## Required behavior

- Do not render the configured Header Image when the exact query value is `iframe=true`.
- Remove the image markup and request. A CSS-only hide does not satisfy this requirement.
- Apply the rule to listing, group, package, and multi-listing booking pages.
- Apply the rule to validation errors, popup checkout pages, and iframe confirmation pages.
- Preserve `iframe=true` when the paid popup opens `/ticket/reserved` in the parent iframe.
- Keep normal public pages, admin pages, and top-level payment callbacks unchanged.
- Keep listing images, themes, Custom CSS, and the iframe-resizer script unchanged.
- Update Header Image help text to state the iframe exception.

## Acceptance criteria

- Iframe HTML contains no `.header-image` element or proxy URL for the configured image.
- The same booking page without `iframe=true` still contains the configured image.
- The paid embedded journey stays in iframe mode through its confirmation page.
- Regression tests cover free and paid embedded journeys.

## References

- `src/ui/templates/layout.tsx`
- `src/shared/iframe.ts`
- `src/features/public/ticket-submit.ts`
- `src/ui/client/admin/checkout-popup.ts`
- `src/locales/en/settings.json`
- `test/ui/templates/layout.test.ts`

### PR DESCRIPTION
**Pull request by stefan-burke**
## What changed

A booking page opened inside another site's iframe (`?iframe=true`) no longer renders the configured Header Image. The paid popup journey keeps iframe mode through its confirmation page.

## Problem

The shared layout rendered the Header Image on every page, including pages embedded in a host site's iframe. The image duplicated the host page's branding and cost an image request. Embed mode already removed the public menu and listing header; the header image remained.

The paid popup had a second defect. After payment, the popup page's client script navigated the iframe to `/ticket/reserved` without `?iframe=true`, so the confirmation page rendered with the normal header again.

## Changes

1. `src/ui/templates/layout.tsx` gates the header image on the per-request iframe mode. `detectIframeMode` already parses the exact query value for every request before routing, so one gate covers listing, group, package, and multi-listing booking pages, validation errors, popup checkout pages, and confirmation pages. No markup and no image request are emitted — this is not a CSS hide.
2. `src/ui/templates/payment.tsx` renders the popup page's confirmation URL through the existing `appendIframeParam` helper as `data-success-href`. The popup page only exists in iframe mode, so the attribute carries `?iframe=true`.
3. `src/ui/client/admin/checkout-popup.ts` navigates the iframe to the rendered URL instead of a hardcoded `/ticket/reserved`.
4. `src/locales/en/settings.json` — the Header Image help text states the iframe exception. `src/locales/en/guide-domains.json` states it in the guide answer.

Normal public pages, admin pages, and top-level payment callbacks are unchanged: the flag is false without the exact query value, so the image renders as before. No stored state changes; the behaviour is read-time only.

## Value

An owner who embeds a booking page on their own site no longer shows duplicated branding inside the iframe. A buyer who pays through the popup sees a confirmation page that stays in iframe mode to the end.

## Tests

- `test/ui/templates/layout.test.ts` — the image is absent in iframe mode; present without the param (existing test).
- `test/ui/templates/payment.test.ts` — the popup page renders `data-success-href="/ticket/reserved?iframe=true"`.
- `test/ui/client/admin/checkout-popup.test.ts` (new) — the client navigates to the rendered URL on `payment-success`, ignores other origins, and keeps the waiting state while the checkout window is open. Full coverage of the module.
- `test/integration/server/public/ticket-slug-get.test.ts` — the booking page hides the image with `?iframe=true` and shows it without.
- `test/integration/server/public/ticket-paid-flow-edge-cases.test.ts` — the paid embedded journey's popup page carries the iframe confirmation URL.

All new tests failed before the fix and pass after it. `deno task precommit` passes.

The mutation runner stalls in this environment (its checkout copy never completed), so the gate ran as three applied mutants, each killed by its mirror test: gate removed, attribute dropped, and the original hardcoded navigation restored.

Closes #2186

## Summary by CodeRabbit

* **Bug Fixes**
  * Checkout completion now navigates correctly when payment pages are embedded in an iframe, preserving the embedded-page context.
  * Header images are hidden on booking pages embedded in an iframe, while remaining visible on standard pages.

* **Documentation**
  * Updated header-image guidance to explain iframe behavior and clarified the file-size limit formatting.
  * Added details about preserving embedded-page context after checkout.

### CODE REVIEW

File: src/locales/en/guide-domains.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
...
+  "guide.a.how_does_the_header_image_work": "<p>Upload a logo or banner from <strong>Settings</strong> and it appears at the top of every admin and public page. A booking page that another site embeds in an iframe hides it, because the host page already shows its own branding. The image is encrypted and served through the <code>/image/</code> proxy with long-lived immutable cache headers, so browsers only download it once.</p><p>Supported formats are JPEG, PNG, and WebP, up to 256 KB. Uploading a new image automatically deletes the old one, and the <strong>Remove Image</strong> button clears it completely. If image storage isn't configured the section is hidden — see <strong>Advanced Settings</strong> to set up a Bunny storage zone.</p>",
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/locales/en/guide-domains.json]: Fixed in 8237ddf30. Both strings now state the exact condition — a booking page hides the image when the page address includes ?iframe=true — and note that the embed code adds it, so an owner embedding by hand knows why the image still shows. The implementation contract (detectIframeMode compares the exact query value in src/shared/iframe.ts) is unchanged; this was copy accuracy only. deno task check:copy, lint, and the header-image settings tests pass.
